src/api/deriveBundleData/deriveChunkGroupData.ts

```diff
@@ -1,36 +1,57 @@
-import { Stats } from '../../types/Stats';
+import { Asset, ChunkAsset, Stats } from '../../types/Stats';
 import { ChunkGroupData } from '../../types/BundleData';
 import { DataOptions } from '../../types/DataOptions';
 
 export function deriveChunkGroupData(stats: Stats, options: DataOptions) {
     const assetFilter = (options && options.assetFilter) || defaultAssetFilter;
     const chunkGroupData: ChunkGroupData = {};
-    const assetSizeMap = getAssetSizeMap(stats);
+
+    const getProcessedAsset = getAssetProcessor(stats);
 
     // Process each named chunk group
     for (let chunkGroupName of Object.keys(stats.namedChunkGroups)) {
         const chunkGroup = stats.namedChunkGroups[chunkGroupName];
 
-        let size = 0;
+        let chunkGroupSize = 0;
         let assets: string[] = [];
         let ignoredAssets: string[] = [];
 
         // Process each asset in the chunk group
         for (let asset of chunkGroup.assets) {
-            if (!assetFilter(asset)) {
-                ignoredAssets.push(asset);
+            const { name, size} = getProcessedAsset(asset);
+            if (!assetFilter(name)) {
+                ignoredAssets.push(name);
             } else {
-                assets.push(asset);
-                size += assetSizeMap.get(asset);
+                assets.push(name);
+                chunkGroupSize += size;
             }
         }
 
-        chunkGroupData[chunkGroupName] = { size, assets, ignoredAssets };
+        chunkGroupData[chunkGroupName] = { size: chunkGroupSize, assets, ignoredAssets };
     }
 
     return chunkGroupData;
 }
 
+function getAssetProcessor(stats: Stats): (asset: ChunkAsset) => Pick<Asset, 'name' | 'size'> {
+    if (
+        Object.values(stats.namedChunkGroups).some(({ assets }) =>
```

Comment:
I don't love all the extra complexity here to handle WP4 and 5.  I get that this is akin to feature detection in a browser, but we're not talking about a matrix with a bunch of browsers x a bunch of versions...it's just v4 and v5.

As I think about it more, I expect that there will be other improvements that we can make if we assume WP5.  I'm not that interested in having to maintain backward compatibility.  My thought is to ditch WP4 support (and bump the major version.)  We can always have a maintenance branch that works against WP4 if we need to do fixes there.  What do you think?  If we go that way then you can simplify a bunch of this.